src/site/constants/environments-configs.js

```diff
@@ -33,11 +33,11 @@ module.exports = {
     BUILDTYPE: ENVIRONMENTS.LOCALHOST,
     BASE_URL: isNode
       ? 'https://localhost:3001'
-      : `http://${location.hostname}${
-          location.port ? `:${location.port}` : ''
+      : `http://${location.hostname || 'localhost'}:${
+          location.port ? location.port : '3001'
         }`,
     API_URL: isNode
       ? 'http://localhost:3000'
-      : `http://${location.hostname}:3000`,
+      : `http://${location.hostname || 'localhost'}:3000`,
```

Comment:
There's some shared DOM setup stuff from the Mocha options. I'm sure we could split out a separate one, but I didn't want to go through the hassle of figuring out what dependencies to include in that and have to maintain it separately. We'd need to know what we require to properly invoke the API request. In any case, it doesn't have to have defaults here.